clientsv1_roles.go

```diff
@@ -0,0 +1,111 @@
+package sams
+
+import (
+	"context"
+	"io"
+	"slices"
+
+	"connectrpc.com/connect"
+	"golang.org/x/oauth2"
+
+	"github.com/google/uuid"
+	clientsv1 "github.com/sourcegraph/sourcegraph-accounts-sdk-go/clients/v1"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/clients/v1/clientsv1connect"
+	"github.com/sourcegraph/sourcegraph-accounts-sdk-go/roles"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// RolesServiceV1 provides client methods to interact with the
+// RolesService API v1.
+type RolesServiceV1 struct {
+	client *ClientV1
+}
+
+func (s *RolesServiceV1) newClient(ctx context.Context) clientsv1connect.RolesServiceClient {
+	return clientsv1connect.NewRolesServiceClient(
+		oauth2.NewClient(ctx, s.client.tokenSource),
+		s.client.gRPCURL(),
+		connect.WithInterceptors(s.client.defaultInterceptors...),
+	)
+}
+
+// RegisterResourcesMetadata is the metadata for a set of resources to be registered.
+type RegisterResourcesMetadata struct {
+	ResourceType roles.ResourceType
+	Revision     uuid.UUID
```

Comment:
I think we should also add equivalently clear docstring here like we did for defining proto about how to use and determine the value of `Revision` and what implication it has (i.e. all resources with a different revision will be deleted after the streaming request is done successfully).

Reply:
Or if possible, maybe we could abstract away from call sites altogether? The SDK always comes up a UUID on its own, since SDK is already taking charge of the entire stream request.